packages/shared/src/components/squads/Details.tsx

```diff
@@ -18,102 +18,109 @@ import { cloudinary } from '../../lib/image';
 import CameraIcon from '../icons/Camera';
 import { formToJson } from '../../lib/form';
 import { Modal } from '../modals/common/Modal';
+import { ModalPropsContext } from '../modals/common/types';
 
 const squadImageId = 'squad_image_file';
 export function SquadDetails({
-  modalState,
   onNext,
   form,
   setForm,
 }: SquadStateProps): ReactElement {
-  if (ModalState.Details !== modalState) return null;
+  const { activeView } = useContext(ModalPropsContext);
+  if (activeView !== ModalState.Details) return null;
   const { name, handle, description } = form;
   const [handleValid, setHandleValid] = useState(true);
-  const onSubmit = async (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    const formJson = formToJson<SquadForm>(e.currentTarget);
-    if (!formJson.name || !formJson.handle) {
-      return;
-    }
-    const handleExists = await checkSourceExists(formJson.handle);
-    setHandleValid(!handleExists);
-    if (!handleExists) {
-      onNext(formJson);
-    }
-  };
+  const onSubmit =
+    (next: () => void) => async (e: FormEvent<HTMLFormElement>) => {
+      e.preventDefault();
+      const formJson = formToJson<SquadForm>(e.currentTarget);
+      if (!formJson.name || !formJson.handle) {
+        return;
+      }
+      const handleExists = await checkSourceExists(formJson.handle);
+      setHandleValid(!handleExists);
+      if (!handleExists) {
+        onNext(formJson);
+        next();
+      }
+    };
   return (
-    <>
-      <Modal.Body>
-        <form
-          className="flex flex-col gap-4 items-center"
-          onSubmit={onSubmit}
-          id="squad-form"
-        >
-          <SquadTitle>
-            Make it look <SquadTitleColor>yours.</SquadTitleColor>
-          </SquadTitle>
-          <ImageInput
-            initialValue={form.file}
-            id={squadImageId}
-            fallbackImage={cloudinary.squads.imageFallback}
-            className={{
-              container: 'rounded-full border-0 my-1',
-              img: 'object-cover',
-            }}
-            hoverIcon={<CameraIcon size="xlarge" />}
-            onChange={(file) => setForm({ file })}
-          />
-          <TextField
-            label="Squad name"
-            inputId="name"
-            name="name"
-            valid={!!name}
-            leftIcon={<UserIcon />}
-            value={name ?? ''}
-            className={{
-              container: 'w-full',
-            }}
-            valueChanged={(text) => setForm({ ...form, name: text })}
-          />
-          <TextField
-            label="Handle"
-            inputId="handle"
-            hint={!handleValid ? 'The handle is already exists' : undefined}
-            valid={handleValid}
```

Comment:
@idoshamun @rebelchris 
Do we need any validation for the handle here? The only validation right now is if it exists or not.

Should we only allow `[a-z0-9-]` ?

Reply:
I think we should add the check on both backend and the frontend